.github/workflows/ci.yml

```diff
@@ -220,6 +221,27 @@ jobs:
           make VERBOSE=1 -j${BUILD_CONCURRENCY} ${make_targets}
           ctest --timeout ${TEST_TIMEOUT} --output-on-failure ${ctest_flags}
 
+  dr-testing:
+    name: Run DR shp-tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Build docker image
+        run: docker build . -t dr-test -f .github/Dockerfile
+      - name: Run tests
+        run: |
+          docker run --name dr-test -d -v $(pwd):/repo dr-test sh -c "
+            cd repo &&
+            mkdir build &&
+            cd build &&
+            cmake -DCMAKE_CXX_STANDARD=23 -DCMAKE_CXX_COMPILER=icpx -DONEDPL_BACKEND=dpcpp -DCMAKE_BUILD_TYPE=Release .. &&
+            cmake --build . --target shp-all-tests -j${nproc} &&
```

Comment:
We might want to use `BUILD_CONCURRENCY` here and below to unify this with the existing tests.

Reply:
ok, applied